enatega-multivendor-app/src/screens/Menu/Menu.js

```diff
@@ -349,13 +397,76 @@ function Menu({ route, props }) {
   }
 
   // Flatten the array. That is important for data sequence
-  const restaurantSections = sections.map(sec => ({
+  const restaurantSections = sectionData?.map(sec => ({
     ...sec,
     restaurants: sec.restaurants
-      .map(id => restaurants.filter(res => res._id === id))
+      .map(id => restaurantData?.filter(res => res._id === id))
       .flat()
   }))
 
+  const extractRating = ratingString => parseInt(ratingString)
+
+  const applyFilters = () => {
+    let filteredData = [...data.nearByRestaurants.restaurants]
+
+    const ratings = filters['Rating']
+    const sort = filters['Sort']
+    const offers = filters['Offers']
+    const cuisines = filters['Cuisines']
+
+    // Apply filters incrementally
+    // Ratings filter
+    if (ratings?.selected?.length > 0) {
+      const numericRatings = ratings.selected.map(extractRating)
+      filteredData = filteredData.filter(
+        item => item?.reviewData?.ratings >= Math.min(...numericRatings)
+      )
+    }
+
+    // Sort filter
+    if (sort?.selected?.length > 0) {
+      if (sort.selected[0] === 'Fast Delivery') {
+        filteredData.sort((a, b) => a.deliveryTime - b.deliveryTime)
+      } else if (sort.selected[0] === 'Distance') {
+        filteredData.sort(
+          (a, b) =>
+            a.distanceWithCurrentLocation - b.distanceWithCurrentLocation
+        )
+      }
+    }
+
+    // Offers filter
+    if (offers?.selected?.length > 0) {
+      if (offers.selected.includes('Free Delivery')) {
+        filteredData = filteredData.filter(item => item?.freeDelivery)
+      }
+      if (offers.selected.includes('Accept Vouchers')) {
+        filteredData = filteredData.filter(item => item?.acceptVouchers)
+      }
+    }
+
+    // Cuisine filter
+    if (cuisines?.selected?.length > 0) {
+      filteredData = filteredData.filter(item =>
+        item.cuisines.some(cuisine => cuisines?.selected?.includes(cuisine))
+      )
+    }
+
+    // Set filtered data
+    setRestaurantData(filteredData)
+
+    // If no filters applied, reset to original data
+    if (
```

Comment:
I dont think so we need this, as we are already assigning filteredData value in the beginning of this function and then only modify if conditions true. In case the condition fails filteredData will always be equal to default value.